runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -656,18 +656,26 @@ private void resurrectCheckpoints() {
         try {
             for (Map.Entry<UUID, StreamMetaData> entry : streamsMetaData.entrySet()) {
                 cfs[i++] = CompletableFuture.runAsync(() -> {
-                    CheckPoint checkPoint = entry.getValue().getLatestCheckPoint();
-                    if (checkPoint == null) {
-                        log.info("resurrectCheckpoints[{}]: Truncated checkpoint for this stream",
-                                Utils.toReadableId(entry.getKey()));
-                        return;
-                    }
-
-                    // For now one by one read and apply
-                    for (long address : checkPoint.getAddresses()) {
-                        updateCorfuObject(getLogData(runtime, loadInCache, address));
-                    }
-                }, executorService);
+                    CheckPoint checkPoint = null;
+                    try {
+                        checkPoint = entry.getValue().getLatestCheckPoint();
+                        if (checkPoint == null) {
+                            log.info("resurrectCheckpoints[{}]: Truncated checkpoint for this stream",
+                                    Utils.toReadableId(entry.getKey()));
+                            return;
+                        }
+
+                        // For now one by one read and apply
+                        for (long address : checkPoint.getAddresses()) {
+                            updateCorfuObject(getLogData(runtime, loadInCache, address));
+                        }
+                    } catch (Throwable t) {
+                        if (checkPoint != null) {
```

Comment:
I think `checkPoint` will not be null here: `CheckPoint checkPoint = entry.getValue().getLatestCheckPoint();` This is a simple operation which should not throw any exception, then the above code checks `if (checkPoint == null)`, which will return. So if we get a Throwable here, it means checkpoint is not null. So I think you can remove this null check.

Reply:
Done.